filebeat/input/filestream/input.go

```diff
@@ -224,17 +224,26 @@ func (inp *filestream) open(log *logp.Logger, canceler input.Canceler, path stri
 // or the file cannot be opened because for example of failing read permissions, an error
 // is returned and the harvester is closed. The file will be picked up again the next time
 // the file system is scanned
-func (inp *filestream) openFile(path string, offset int64) (*os.File, error) {
-	err := inp.checkFileBeforeOpening(path)
+func (inp *filestream) openFile(log *logp.Logger, path string, offset int64) (*os.File, error) {
+	f, err := os.OpenFile(path, os.O_RDONLY, os.FileMode(0))
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed opening %s: %s", path, err)
 	}
```

Comment:
nit; now you are missing `f.Close()` on a few errors.

I would recommend to move all code after OpenFile into a separate function, or use this pattern (both strategies help to not forget to Close the file if an error occurs in the future):

```
f, err := os.OpenFile(...)
if err != nil { ... }

var ok bool
defer cleanup.IfNot(&ok, cleanup.IgnoreError(f.Close))

// checks
...

// final return
ok = true
return f, nil
```